mutation/partition_version_list.hh

```diff
@@ -8,7 +8,6 @@
 
 #pragma once
 
-#include "utils/assert.hh"
```

Comment:
i am not sure if this is the best fix. there are two options i think

1. to run the linter tool in Debug mode: i implemented this in #21739
2. guard this include with `SEASTAR_DEBUG` like

   ```c++
   #ifdef SEASTAR_DEBUG
   #include "utils/assert.hh"
   #endif
   ```

   this could be more correct. because, in this header, we do use `SCYLLA_ASSERT()`, which is in turn defined by `utils/assert.hh`. and a header is supposed to be self-contained. so we do need to include it. but this header is only used in the debug build.

what do you think?

Reply:
I suppose changing the linter to use the debug should be enough ... although the usage of `SCYLLA_ASSERT()` in these files looks to be guarded by the `SEASTAR_DEBUG` indeed so it would make sense to also guard the inclusion.